sympy/core/mul.py

```diff
@@ -1255,19 +1255,20 @@ def _eval_is_integer(self):
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
```

Comment:
perhaps all that was needed was a `return` if `_self == self`

Reply:
Ln 1259 ☝️ 
check out this comment https://github.com/sympy/sympy/issues/19174#issuecomment-618611992
```python3
(Pdb) p n/d
*** RecursionError: maximum recursion depth exceeded while calling a Python object
```

Reply:
ahh, ok -- so don't rebuild object (which might check assumptions while doing so)?

Reply:
I tracked it here: https://github.com/sympy/sympy/issues/19174#issuecomment-618706618

Reply:
Yes. I am just commenting that this should be a general principle for these routines: don't put parts back together.